sycl/source/detail/queue_impl.hpp

```diff
@@ -98,6 +103,31 @@ class queue_impl {
         MIsProfilingEnabled(has_property<property::queue::enable_profiling>()),
         MHasDiscardEventsSupport(MDiscardEvents &&
                                  (MHostQueue ? true : MIsInorder)) {
+    // The following commented section provides a guideline on how to use the
+    // TLS enabled mechanism to create a tracepoint and notify using XPTI. This
+    // is the prolog section and the epilog section will initiate the
+    // notification.
+    // #if XPTI_ENABLE_INSTRUMENTATION
+    //     /// This section of code is relying on scoped objects, so they cannot
+    //     be
+    //     /// encapsulated in a function
+    //     detail::tls_code_loc_t Tls;
+    //     auto TData = Tls.query();
+    //     xpti::framework::tracepoint_t TP(TData.fileName(),
+    //     TData.functionName(),
+    //                                      TData.lineNumber(),
+    //                                      TData.columnNumber    (), (void
+    //                                      *)this);
+    //     if (xptiTraceEnabled()) {
+    //       TP.stream(SYCL_STREAM_NAME)
+    //           .trace_type(xpti::trace_point_type_t::queue_create);
+    //       // If the member variables in the class are uncommented or breaking
+    //       binary
+    //       // compatibility is allowed, then TP.stream_id(), TP.trace_event()
+    //       and
+    //       // TP.instance_id() can be cached in the member variables
```

Comment:
It shouldn't be an ABI break if new member variables are added to the `queue_impl` class.

Reply:
Okay, I assumed class layout was affected - have updated it to cache it in the class now.

Reply:
Okay, caching the information in member variables now.